components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -309,6 +310,74 @@ void ConversationDriver::OnConversationActiveChanged(
   DVLOG(3) << "Conversation active changed: " << is_conversation_active;
   MaybeSeedOrClearSuggestions();
   MaybePopPendingRequests();
+  MaybeFetchSearchQuerySummary();
+}
+
+void ConversationDriver::ClearSearchQuerySummary() {
+  // Only clear it when the conversation only has the staged query and summary.
+  if (chat_history_.size() != 2) {
+    return;
+  }
+
+  const auto& last_turn = chat_history_.back();
+  if (last_turn->events && !last_turn->events->empty() &&
+      last_turn->events->back()->is_completion_event() &&
+      last_turn->events->back()
+          ->get_completion_event()
+          ->from_brave_search_SERP) {
+    chat_history_.pop_back();  // summary
+    chat_history_.pop_back();  // query
```

Comment:
extremely optional:

```suggestion
chat_history_.resize(0);
```

Reply:
Use clear() in https://github.com/brave/brave-core/pull/25257/commits/203468ddb1a948ddb7a5767547c41426efcffe14.

Reply:
lol that's much more sensible :laughing: